### PR TITLE
Using ArrayPool in RenderData

### PR DESCRIPTION
**Pull request by lindexi**
## Description

Using ArrayPool in RenderData. The max length is 72, see MILCMD_DRAW_ROUNDED_RECTANGLE_ANIMATE

https://github.com/dotnet/wpf/blob/89d172db0b7a192de720c6cfba5e28a1e7d46123/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Generated/RenderData.cs#L149-L184

## Customer Impact

None. But it will affect the rendering of all DrawingVisual and UIElement 

## Regression

dotnet 7

## Testing

Just CI and my [demo](https://github.com/lindexi/lindexi_gd/commit/d40a0dd82fb7d6bf8fa24b4b4206b968ec44a07d)

## Risk

Low

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/RenderData.cs [hunk 1]
```diff
@@ -52,6 +53,11 @@ internal RenderData()
             CanBeInheritanceContext = false;
         }
 
+        ~RenderData()
+        {
+            ArrayPool<byte>.Shared.Return(_buffer);
```
Comment (gfoidl) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/RenderData.cs:58]: Can this just be "disposed" normally? Using a finalizer is quite heavy and puts pressure on the GC's finalizer queue. Especially with 

> The max length is 72

in mind, maybe it's just cheaper to allocate and let the GC clean up. 
Or as there is unsafe code already involved use a struct with fixed sized buffer of that size, and wrap by a span for access outside that class?

Comment (lindexi) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/RenderData.cs]: @gfoidl  Thank you and I update the PR

**Comment by kronic** [comment]
@lindexi  maybe you should delete the field altogether _buffer

**Comment by lindexi** [comment]
@kronic Thank you. But I find it hard to delete the `_buffer` field

**Comment by omariom** [comment]
@dipeshmsft 
What does the team use for perf benchmarks?

**Comment by dipeshmsft** [comment]
@omariom  We mainly use sample applications to review the performance benefits and we rely on our regression suite to check any performance regressions.

**Comment by lindexi** [comment]
@dipeshmsft Thank you.

**Comment by omariom** [comment]
@dipeshmsft  
Can their source be opened so we could use them in out PRs?

**Comment by dipeshmsft** [comment]
@omariom , we are working on open-sourcing the complete test-suite. You can check the repo [dotnet/wpf-test](https://github.com/dotnet/wpf-test) , but the task is still in progress. @gurpreet-wpf

**Comment by lindexi** [comment]
@dipeshmsft  Can I publish my private WPF framework version?

**Comment by dipeshmsft** [comment]
@vishalmsft @gurpreet-wpf Could you take a look at this ??

**Comment by vishalmsft** [comment]
@lindexi do you mean tests written and available in your private repo?

**Comment by lindexi** [comment]
@vishalmsft I mean that can I publish my nuget package which include my private WPF framework. Some of my friends can't wait for the official WPF release. I expect to provide a private version for them to use.

**Comment by lindexi** [comment]
And this version wpf framework may used in the business application.

**Pull request merged**